dev/AppNotifications/WICUtility.cpp

```diff
@@ -0,0 +1,165 @@
+    winrt::com_ptr<IWICImagingFactory> const& wicImagingFactory,
+    winrt::com_ptr<IWICBitmapSource> const& wicBitmapSource,
+    _In_ REFGUID guidContainerFormat,
+    _In_ BITMAP_VERSION bmpv)
+{
+    winrt::com_ptr<IStream> spImageStream;
+    THROW_IF_FAILED(CreateStreamOnHGlobal(nullptr /* handle */, true /* delete on release */, spImageStream.put()));
+
+    // Create encoder and initialize it
+    winrt::com_ptr<IWICBitmapEncoder> wicEncoder;
+    THROW_IF_FAILED(wicImagingFactory->CreateEncoder(guidContainerFormat, nullptr, wicEncoder.put()));
+
+    THROW_IF_FAILED(wicEncoder->Initialize(spImageStream.get(), WICBitmapEncoderCacheOption::WICBitmapEncoderNoCache));
+
+    // Add a single frame to the encoder with the Bitmap
+    AddFrameToWICBitmap(wicImagingFactory, wicEncoder, wicBitmapSource, bmpv);
+
+    THROW_IF_FAILED(wicEncoder->Commit());
+
+    // Seek the stream to the beginning and transfer
+    static const LARGE_INTEGER lnBeginning = {};
+    THROW_IF_FAILED(spImageStream->Seek(lnBeginning, STREAM_SEEK_SET, nullptr /* new seek pointer */));
+
+    return spImageStream;
+}
+
+void SaveImageWithWIC(
+    winrt::com_ptr<IWICImagingFactory> const& wicImagingFactory,
+    winrt::com_ptr<IWICBitmapSource> const& wicBitmapSource,
+    _In_ REFGUID guidContainerFormat,
+    winrt::com_ptr<IStream>& pStream)
+{
+    winrt::com_ptr<IStream> spImageStream;
+    spImageStream = GetStreamOfWICBitmapSource(wicImagingFactory, wicBitmapSource, guidContainerFormat, BMPV_1);
+
+    // Seek the stream to the beginning and transfer
+    static LARGE_INTEGER const lnBeginning{ 0 };
+    THROW_IF_FAILED(spImageStream->Seek(lnBeginning, STREAM_SEEK_SET, nullptr /* new seek pointer */));
+
+    static ULARGE_INTEGER lnbuffer{ INT_MAX };
+    THROW_IF_FAILED(spImageStream->CopyTo(pStream.get(), lnbuffer, nullptr /* pointer to number of bytes read */, nullptr /* pointer to number of bytes written */));
+}
+
+void Microsoft::Windows::AppNotifications::WICHelpers::WriteHIconToPngFile(wil::unique_hicon const& hIcon, _In_ PCWSTR pszFileName)
+{
+    auto wicImagingFactory{ winrt::create_instance<IWICImagingFactory>(CLSID_WICImagingFactory, CLSCTX_INPROC_SERVER) };
+   
+    winrt::com_ptr<IWICBitmap> wicBitmap;
+    THROW_IF_FAILED(wicImagingFactory->CreateBitmapFromHICON(hIcon.get(), wicBitmap.put()));
+
+    // Create stream to save the HICON to.
+    winrt::com_ptr<IStream> spStream;
+    THROW_IF_FAILED(CreateStreamOnHGlobal(nullptr /* handle */, TRUE, spStream.put()));
+
+    winrt::com_ptr<IWICBitmapSource> wicBitmapSource;
+    wicBitmapSource = wicBitmap;
+
+    SaveImageWithWIC(wicImagingFactory, wicBitmapSource, GUID_ContainerFormatPng, spStream);
+
+    // Write out the stream to a file.
+    winrt::com_ptr<IStream> spStreamOut;
+    THROW_IF_FAILED(SHCreateStreamOnFileEx(pszFileName, STGM_CREATE | STGM_WRITE | STGM_SHARE_DENY_WRITE, FILE_ATTRIBUTE_NORMAL, TRUE, nullptr, spStreamOut.put()));
+
+    STATSTG statstg;
```

Comment:
Always ensure variables are initialized. Any not-class (i.e. anything without a default constructor) isn't initialized unless you help

```c++
STATSTG statstg{};
```

Same comment applies throughout